docs/savegame_format.md

```diff
@@ -184,3 +184,31 @@ We advise you to call this setting `__mypp_auto_destroy_rivers` in the settings
 Doing it this way ensures that a savegame created by these patch-packs can still safely be loaded by unpatched clients.
 They will simply ignore the field and continue loading the savegame as usual.
 The prefix is strongly advised to avoid conflicts with future-settings in an unpatched client or conflicts with other patch-packs.
+
+## Scripts custom data format
+
+Script chunks (`AIPL` and `GSDT`) use `CH_TABLE` chunk type.
+
+At the end of each record there's an `SLE_FILE_U8` to indicate if there's custom data (1) or not (0).
+
+There are 6 data types saved for scripts.
+When saving, each data starts with the type marker saved as `SLE_FILE_U8` followed by the actual data.
+- SQLS_INT (0):
+  - an `SLE_FILE_I64` with the actual value (`SLE_FILE_I32` before savegame version 296)
+- SQLS_STRING (1):
+  - an `SLE_FILE_U8` with the string length
+  - string len `SLE_FILE_I8`
+- SQLS_ARRAY (2):
+  - each element saved as a separate data (ie type then value)
+  - an SQLS_ARRAY_TABLE_END (0xFF) marker (`SLE_FILE_U8`)
+- SQSL_TABLE (3):
+  - for each element:
+    - key saved as a separate data
```

Comment:
here too, it is not clear that you mean this data-type with "separate data". Maybe just name it. Start this blob with:

`There are 6data types for scripts, called script-data-type`.

After that you can just say: `key saved as script-data-type`

Invalidates the above comment ofc ;)